src/BartIO.jl

```diff
@@ -23,27 +49,59 @@ export wrapper_bart
     bart(1,"phantom -k -x128")
     ````
 """
-function wrapper_bart(pathtobart::String)
-    if !isdir(pathtobart)
-        @warn "BART folder does not exists"
+function bart(nargout::Int,cmd,args::Vararg{Array{ComplexF32}})
+    # Check input variables
+    if isdispatchtuple(args) || isempty(cmd) nargout < 0
+        @warn "Usage: bart(<nargout>,<command>, <arguments...>\n\n"
+        return nothing
+    end
+
+    # Check bart toolbox path
+    bart_path = get_bart_path()
+    if isempty(bart_path)
+        @error "BART path not detected.\n Use : `set_bart_path(pathToBart)`"
+    end
+
+    nargin = length(args)
+    name = mktempdir(tempdir(); prefix="jl_", cleanup=true)
+    infiles = [name * "/in" * string(idx) for idx in 1:nargin]
+
+    for idx in 1:nargin
+        write_cfl(infiles[idx], args[idx])
     end
 
-    python_pycall = PyCall.python
+    outfiles = [name*"/out"*string(idx) for idx in 1:nargout]
 
-    run(`$python_pycall -m pip install numpy`)
+    shell_cmd = bart_path*"/bart"
+    cmd_split = split(cmd)
 
-    PyCall.py"""
-    import os
-    import sys
-    os.environ['TOOLBOX_PATH'] = $pathtobart
-    path = os.environ["TOOLBOX_PATH"] + "/python/"
-    sys.path.append(path)
-    """
+    run(`$shell_cmd $cmd_split $infiles $outfiles`)
+
+    output = Array{ComplexF32}[]
```

Comment:
You already know the size of `output`, right? In this case, I think it is better to define this as something like `Vector{Array{ComplexF32}}(undef, length(output))` and then  index in `output[idx]` in the for loop.